core/lensQueryHelpers.ml

```diff
@@ -0,0 +1,470 @@
+open Types
+open Sugartypes
+open Operators
+open Utility
+open LensUtility
+open Value
+open LensRecordHelpers
+
+exception Runtime_error = Errors.Runtime_error
+
```

Comment:
This overlaps to some extent with query.ml and queryshredding.ml, except that you are generating set-valued queries instead of multiset-valued ones.  I don't suggest any change for the time being but at some point we should extend the core query language with duplicate elimination and translate the RL-related queries to that instead of having two (or more) very similar query generators.